### ISSUE TITLE
Support for local test runners

### ISSUE THREAD

**Issue by nicojs** [stryker-mutator/stryker#137]
Lets say I have a project with a `stryker.conf.js` in the root and i want to use a custom test runner `my-runner`.

I should be able to do this:

``` javascript
// myTestRunner.js
var TestRunnerFactory = require('stryker-api').TestRunnerFactory;

function MyTestRunner(options){
  this.options = options;
}

MyTestRunner.prototype.run = function(options){
  console.log('run!', options);
  return Promise.resolve();
}

TestRunnerFactory.instance().register('my-runner', MyTestRunner);
```

``` javascript
// stryker.conf.js
module.exports = function(config){
  config.set({
     //...
     testRunner: 'my-runner',
    plugins: ['./myTestRunner']
  });
}
```

However when I run stryker (with `logLevel: 'trace'`)i get the following error:

``` bash
[2016-07-28 17:38:42.604] [TRACE] IsolatedTestRunnerAdapter - [2016-07-28 17:38:42.604] [WARN] PluginLoader - Cannot find plugin "./myTestRunner".

  Did you forget to install it ?
  npm install ./myTestRunner --save-dev
...
Error: Could not find a testrunner with name my-runner, did you install it correctly (for example: npm install --save-dev stryker-my-runner-runner)?
```

This is because the test runner runs in a child process. It uses the PluginLoader correctly, but it uses a different working directory from the project root.

To solve this, I think we should fix this by passing the cwd of the root process and prepending it to any local plugin (plugins starting with a `.`).

Further more, i could also add the test runner in the `stryker.conf.js` file it self:

``` javascript
// stryker.conf.js
var TestRunnerFactory = require('stryker-api').TestRunnerFactory;

function MyTestRunner(options){ /*...*/ }

MyTestRunner.prototype.run = function(options){ /*...*/}

TestRunnerFactory.instance().register('my-runner', MyTestRunner);

module.exports = function(config){
  config.set({
    // ...
    testRunner: 'my-runner'
  });
}
```

However, if we want to support this scenario, we should also make sure we import the config file itself from the child process the test runner starts in, just to make sure we also get all side effects. Do we want that? For example [karma](http://karma-runner.github.io/1.0/config/plugins.html) allows you to _inline_ plugins.

I would say we could leave that out for now.

**Comment by simondel** [comment]
I think that a config should just be a config and not contain code for a test runner. I think it's a good idea to take every plugin that starts with a `.` as a relative path and try to include it that way.

What would be bettter, to take the current working directory as the starting point to look or the location of the config file?

**Comment by nicojs** [comment]
I think the location of the config file is the most logical. That way, it would be as if you `require` it from the config file. That should be the case for every plugin.

For example, if i put my config in a different directory:

```typescript
// config/stryker.conf.js
module.exports = function(config){
  config.set({
     //...
     reporter: 'my-reporter',
    plugins: ['./MyReporter']
  });
}
```

My reporter is in the same directory:

```typescript
// config/MyReporter.js
var ReporterFactory = require('stryker-api').ReporterFactory;
ReporterFactory.instance().register('my-reporter', function(){});
```

I think this now results in an error. This should be fixed with this issue as well.

**Comment by nicojs** [comment]
Close for now (will be fixed with #650)
